src/main/java/com/questhelper/spells/Spell.java

```diff
@@ -0,0 +1,144 @@
+	TROLLHEIM_TELEPORT(49, 61, b -> b.rune(2, LAW).rune(2, FIRE)), // TODO: QUEST REQ -> EADGARS RUSE QUEST
+	WIND_WAVE(50, 62, b -> b.rune(BLOOD).rune(5, AIR)),
+	CHARGE_FIRE_ORB(51, 63, b -> b.rune(3, COSMIC).rune(30, FIRE).item(ItemID.UNPOWERED_ORB)),
+	APE_ATOLL_TELEPORT(52, 64, b -> b.rune(2, LAW).rune(2, WATER).rune(2, FIRE).item(ItemID.BANANA)),
+	WATER_WAVE(53, 54, b -> b.rune(BLOOD).rune(5, AIR).rune(7, WATER)),
+	CHARGE_AIR_ORB(54, 66, b -> b.rune(3, COSMIC).rune(30, AIR).item(ItemID.UNPOWERED_ORB)),
+	VULNERABILITY(55, 66, b -> b.rune(SOUL).rune(5, WATER).rune(5, EARTH)),
+	ENCHANT_LVL_5(56, 68, b -> b.rune(COSMIC).rune(15, WATER).rune(15, EARTH)),
+	ENCHANT_DRAGONSTONE_BOLT(8, 68, b -> b.rune(SOUL).rune(COSMIC).rune(12, EARTH)),
+	KOUREND_CASTLE_TELEPORT(57, 69, b -> b.rune(2, SOUL).rune(2, LAW).rune(4,WATER).rune(5, FIRE)), //TODO: UNLOCKED VIA BOOK
+	EARTH_WAVE(58, 70, b -> b.rune(BLOOD).rune(5, AIR).rune(7, EARTH)),
+	ENFEEBLE(59, 73, b -> b.rune(SOUL).rune(8, WATER).rune(8, EARTH)),
+	TELEOTHER_LUMBRIDGE(60, 74, b -> b.rune(SOUL).rune(LAW).rune(EARTH)),
+	FIRE_WAVE(61, 75, b -> b.rune(BLOOD).rune(5, AIR).rune(7, FIRE)),
+	ENTANGLE(62, 79, b -> b.rune(4, NATURE).rune(5, WATER).rune(5, EARTH)),
+	STUN(63, 80, b -> b.rune(SOUL).rune(12, WATER).rune(12, EARTH)),
+	CHARGE(64, 80, b -> b.rune(3, BLOOD).rune(3, FIRE).rune(3, AIR)),
+	WIND_SURGE(65, 81, b -> b.rune(WRATH).rune(7, AIR)),
+	TELEOTHER_FALADOR(66, 82, b -> b.rune(SOUL).rune(LAW).rune(WATER)),
+	WATER_SURGE(67, 85, b -> b.rune(WRATH).rune(7, AIR).rune(10, WATER)),
+	TELE_BLOCK(68, 85, b -> b.rune(LAW).rune(DEATH).rune(CHAOS)), //TODO: IN WILDERNESS
+	TELEPORT_TO_TARGET(69, 85, b -> b.rune(LAW).rune(DEATH).rune(CHAOS)), //TODO: HAVE READ TARGET TELEPORT SCROLL
+	ENCHANT_LVL_6(70, 87, b -> b.rune(COSMIC).rune(20, FIRE).rune(20, EARTH)),
+	ENCHANT_ONYX_BOLT(8, 87, b -> b.rune(DEATH).rune(COSMIC).rune(20, FIRE)),
+	TELEOTHER_CAMELOT(71, 90, b -> b.rune(LAW).rune(2, SOUL)),
+	EARTH_SURGE(72, 90, b -> b.rune(WRATH).rune(7, AIR).rune(10, EARTH)),
+	ENCHANT_LVL_7(73, 93, b -> b.rune(COSMIC).rune(20, SOUL).rune(20, BLOOD)),
+	FIRE_SURGE(74, 95, b -> b.rune(WRATH).rune(7, AIR).rune(10, FIRE)),
+	;
+
+	private final int groupId;
+	private final int widgetID;
+	private final int requiredMagicLevel;
+	private final UnaryOperator<StandardSpellBuilder> operator;
+	Spell(int widgetID, int requiredMagicLevel)
+	{
+		this.groupId = 218;
+		this.widgetID = widgetID;
+		this.requiredMagicLevel = requiredMagicLevel;
+		this.operator = UnaryOperator.identity(); // Spell has no requirements
+	}
+
+	Spell(int widgetID, int requiredMagicLevel, UnaryOperator<StandardSpellBuilder> operator)
+	{
+		this.groupId = 218;
+		this.widgetID = widgetID;
+		this.requiredMagicLevel = requiredMagicLevel;
+		this.operator = operator;
+	}
+
+	public Requirement getRequirements()
+	{
+		return operator.apply(StandardSpellBuilder.builder(this)).build();
```

Comment:
I left it as Requirement for now since I'll be doing more work on this. It will return the specific type of requirement that is decided upon.